src/hardware/gameblaster.cpp

```diff
@@ -126,53 +200,39 @@ static uint8_t read_cms_detect(io_port_t port, io_width_t)
 	return retval;
 }
 
-class CMS final : public Module_base {
-private:
-	IO_WriteHandleObject WriteHandler = {};
-	IO_WriteHandleObject DetWriteHandler = {};
-	IO_ReadHandleObject DetReadHandler = {};
-
-public:
-	CMS(Section *configuration) : Module_base(configuration)
-	{
-		Section_prop * section = static_cast<Section_prop *>(configuration);
-		Bitu sampleRate = section->Get_int( "oplrate" );
-		cmsBase = static_cast<io_port_t>(section->Get_hex("sbbase"));
-		WriteHandler.Install(cmsBase, write_cms, io_width_t::byte, 4);
-
-		// A standalone Gameblaster has a magic chip on it which is
-		// sometimes used for detection.
-		const char * sbtype=section->Get_string("sbtype");
-		if (!strcasecmp(sbtype,"gb")) {
-			DetWriteHandler.Install(cmsBase + 4u, write_cms_detect, io_width_t::byte, 12);
-			DetReadHandler.Install(cmsBase, read_cms_detect, io_width_t::byte, 16);
-		}
+void GameBlaster::Close()
+{
+	if (!is_open)
+		return;
 
-		/* Register the Mixer CallBack */
-		cms_chan = MIXER_AddChannel(CMS_CallBack, sampleRate, "CMS");
+	const auto card_name = is_standalone_gameblaster ? "GAMEBLASTER" : "CMS";
```

Comment:
Duplicate of line 33. Extract to function, or at least extract the string constants?

Reply:
Extracted to function.